src/Presentation/ListFacetHtmlBuilder.php

```diff
@@ -98,19 +98,33 @@ private function buildCheckboxesViewModel( FacetConfig $config, PropertyConstrai
 
 		$selectedValues = $combineWithAnd ? $state->getAndSelectedValues() : $state->getOrSelectedValues();
 
-		$checkboxes = [];
+		$visibleCheckboxes = [];
+		$collapsedCheckboxes = [];
 
 		foreach ( $this->getValuesAndCounts( $config ) as $i => $valueCount ) {
-			$checkboxes[] = [
+			$checkbox = [
 				'label' => $valueCount->value,
 				'count' => $valueCount->count,
 				'checked' => in_array( $valueCount->value, $selectedValues ), // TODO: test with multiple types of values
 				'value' => $valueCount->value,
 				'id' => $state->propertyId->getSerialization() . "-$i",
 			];
+
```

Comment:
With the additional logic in this function, it becomes appealing to extract that body of the loop into a ne function